packages/parameters/src/base/BaseProvider.ts

```diff
@@ -44,27 +45,25 @@ abstract class BaseProvider implements BaseProviderInterface {
   public constructor({
     awsSdkV3Client,
     clientConfig,
-    proto,
-  }: {
-    awsSdkV3Client?: unknown;
-    clientConfig?: unknown;
-    proto: new (config?: unknown) => unknown;
-  }) {
+    awsSdkV3ClientPrototype,
+  }: BaseProviderConstructorOptions) {
     this.store = new Map();
     this.envVarsService = new EnvironmentVariablesService();
     if (awsSdkV3Client) {
-      if (!isSdkClient(awsSdkV3Client)) {
+      if (!isSdkClient(awsSdkV3Client) && awsSdkV3ClientPrototype) {
```

Comment:
Is there a chance that `awsSdkV3Client` is v2 and the caller doesn't pass the `awsSdkV3ClientPrototype` parameter to this constructor? If so, this will end the upgrade using a v2 client on line 59, which isn't desired, no?

I can be wrong and this could be being passed somewhere in the code, but I didn't see it.

Reply:
We don't support v2 client at all